src/hints/execution.rs

```diff
@@ -326,3 +361,409 @@ pub fn exit_call(
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let tx = exec_scopes.get::<InternalTransaction>("tx")?;
+    let len = tx.calldata.unwrap_or_default().len();
+    insert_value_into_ap(vm, Felt252::from(len))
+}
+
+pub const TX_CALLDATA: &str = "memory[ap] = to_felt_or_relocatable(segments.gen_arg(tx.calldata))";
+
+pub fn tx_calldata(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    _ids_data: &HashMap<String, HintReference>,
+    _ap_tracking: &ApTracking,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let tx = exec_scopes.get::<InternalTransaction>("tx")?;
+    let calldata = tx.calldata.unwrap_or_default().iter().map(|felt| felt.into()).collect();
+    let calldata_base = vm.add_memory_segment();
+    vm.load_data(calldata_base, &calldata)?;
+    insert_value_into_ap(vm, calldata_base)
+}
+
+pub const TX_ENTRY_POINT_SELECTOR: &str = "memory[ap] = to_felt_or_relocatable(tx.entry_point_selector)";
+pub fn tx_entry_point_selector(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    _ids_data: &HashMap<String, HintReference>,
+    _ap_tracking: &ApTracking,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let tx = exec_scopes.get::<InternalTransaction>("tx")?;
+    let entry_point_selector = tx
+        .entry_point_selector
+        .ok_or(HintError::CustomHint("tx.entry_point_selector is None".to_string().into_boxed_str()))
+        .unwrap_or_default();
+    insert_value_into_ap(vm, entry_point_selector)
+}
+
+pub const RESOURCE_BOUNDS: &str = indoc! {r#"
+    from src.starkware.starknet.core.os.transaction_hash.transaction_hash import (
+        create_resource_bounds_list,
+    )
+
+    ids.resource_bounds = (
+        0
+        if tx.version < 3
+        else segments.gen_arg(create_resource_bounds_list(tx.resource_bounds))
+    )"#
+};
+
+pub fn resource_bounds(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let tx = exec_scopes.get::<InternalTransaction>("tx")?;
+    let version = tx.version.unwrap_or_default();
+    assert!(version < 3.into(), "tx.version >= 3 is not supported yet");
+
+    // TODO: implement resource_bounds for tx.version >= 3
+    // let resource_bounds = if tx.version < 3 {
+    //     0
+    // } else {
+    //     let resource_bounds = tx.resource_bounds.unwrap_or_default().iter().map(|felt|
+    // felt.into()).collect();     let resource_bounds_base = vm.add_memory_segment();
+    //     vm.load_data(resource_bounds_base, &resource_bounds)?;
+    //     resource_bounds_base
+    // };
+
+    let resource_bounds = 0;
+    insert_value_from_var_name("resource_bounds", resource_bounds, vm, ids_data, ap_tracking)
+}
+
+// copilot prompt:
+// implement rust hints for the following python hints:
```

Comment:
You mean we will forget about it?

Reply:
Yeah, I imagine it getting forgotten about, getting stale, and getting out of sync with the hints it is about, and I think an issue about it could be a good way to make sure we revisit it later.

Reply:
We won't forget it because we know we need to support transactions v3 and we have a TODOs that are easy to track:
![image](https://github.com/keep-starknet-strange/snos/assets/190855/dede08ed-0b05-4fee-bddc-650b301ae06c)
.